### PR TITLE
Unit test fix for non US-locale environments

### PR DESCRIPTION
**Pull request by rraomsft**
Using DateTime.UtcNOw and roundtrip date formats for mock JSON.

### CODE REVIEW

**Comment by rraomsft** [comment]
#396

**Comment by rohita-msft** [comment]
lgtm

**Comment by kapil-ms** [comment]
lgtm

File: UnitTests/Infrastructure/DeviceTelemetryRepositoryTest.cs [hunk 1]
```diff
@@ -43,12 +43,11 @@ public DeviceTelemetryRepositoryTest()
             var minTime = new DateTime(year, month, date);
 
             var blobReader = new Mock<IBlobStorageReader>();
-            var blobData =
-                "deviceid,temperature,humidity,externaltemperature,eventprocessedutctime,partitionid,eventenqueuedutctime,IoTHub" +
+            var blobData = "deviceid,temperature,humidity,externaltemperature,eventprocessedutctime,partitionid,eventenqueuedutctime,IoTHub" +
                 Environment.NewLine +
-                "test1,34.200411299709423,32.2233033525866,,2016 - 08 - 04T23: 07:14.2549606Z,3," + minTime + ",Record";
+                "test1,34.200411299709423,32.2233033525866,,2016 - 08 - 04T23: 07:14.2549606Z,3," + minTime.ToString("o") + ",Record";
```
Comment (networkfusion) [inline UnitTests/Infrastructure/DeviceTelemetryRepositoryTest.cs]: although not strictly necessary in the unit test `2016 - 08 - 04T23: 07:14.2549606Z` is invalid. Consider removing the spaces...

**Pull request merged**